src/core/unit.c

```diff
@@ -3317,6 +3393,10 @@ int unit_set_slice(Unit *u, Unit *slice, UnitDependencyMask mask) {
         if (UNIT_GET_SLICE(u) && u->cgroup_realized)
                 return -EBUSY;
 
+        /* Only allow setting one slice; remove previous ones if they exist */
+        if (UNIT_GET_SLICE(u) && UNIT_GET_SLICE(u) != slice)
+                unit_remove_dependencies_by_type(u, UNIT_IN_SLICE);
+
```

Comment:
`UNIT_GET_SLICE(u) != slice` is unnecessary here (given the bail out on `UNIT_GET_SLICE(u) == slice` above.